extras/docker/Dockerfile

```diff
@@ -1,10 +1,11 @@
-FROM python:3.10-slim as base
+FROM python:3.11-slim as base
```

Comment:
I would also agree with @security-companion here that the Docker and `dependency_check` changes belong in a separate PR.